### PR TITLE
[Wsign-compare] warning resolutions, by directory, 4

### PR DESCRIPTION
**Pull request by tg-at-google**
Resolution affecting directories:
`tensorflow/compiler/tf2xla`

### CODE REVIEW

**Comment by tg-at-google** [comment]
@mihaimaruseac

File: tensorflow/compiler/tf2xla/functionalize_cond.cc [hunk 1]
```diff
@@ -224,7 +224,8 @@ string DebugString(const CondArgNodes& nodes) {
 }
 
 StateMap::CondId StateMap::LookupCondId(const Node* node) const {
-  if (node->id() < node_to_condid_map_.size())
+  const int64  node_to_condid_map_size =  node_to_condid_map_.size();
+  if (node->id() < node_to_condid_map_size)
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/functionalize_cond.cc:228]:
```suggestion
  const int64  map_size =  node_to_condid_map_.size();
  if (node->id() < map_size)
```

File: tensorflow/compiler/tf2xla/functionalize_cond.cc [hunk 2]
```diff
@@ -235,14 +236,16 @@ StateMap::CondId StateMap::GetCondId(const StateMap::CondState& state) {
 }
 
 void StateMap::ResetCondId(const Node* node, StateMap::CondId id) {
-  if (node->id() < node_to_condid_map_.size())
+  const int64 node_to_condid_map_size = node_to_condid_map_.size(); 
+  if (node->id() < node_to_condid_map_size)
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/functionalize_cond.cc:240]: Same as above. These locals are short lived, they are declared immediately before they are used so we don't need to be overly verbose.

File: tensorflow/compiler/tf2xla/functionalize_cond.cc [hunk 3]
```diff
@@ -235,14 +236,16 @@ StateMap::CondId StateMap::GetCondId(const StateMap::CondState& state) {
...
     node_to_condid_map_[node->id()] = id;
   else
     added_node_condid_mapping_[node->id()] = id;
 }
 
 StateMap::AncestorId StateMap::LookupAncestorId(const Node* node) const {
-  if (node->id() < node_to_ancestorid_map_.size())
+  const int64 node_to_ancestorid_map_size = node_to_ancestorid_map_.size();
+  if (node->id() < node_to_ancestorid_map_size)
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/functionalize_cond.cc:248]: Same as above. These locals are short lived, they are declared immediately before they are used so we don't need to be overly verbose.

File: tensorflow/compiler/tf2xla/functionalize_cond.cc [hunk 4]
```diff
@@ -254,7 +257,8 @@ StateMap::AncestorId StateMap::GetAncestorId(
 }
 
 void StateMap::ResetAncestorId(const Node* node, StateMap::AncestorId id) {
-  if (node->id() < node_to_ancestorid_map_.size())
+  const int64 node_to_ancestorid_map_size = node_to_ancestorid_map_.size();
+  if (node->id() < node_to_ancestorid_map_size)
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/functionalize_cond.cc:261]: Same as above. These locals are short lived, they are declared immediately before they are used so we don't need to be overly verbose.

File: tensorflow/compiler/tf2xla/rearrange_function_argument.cc [hunk 5]
```diff
@@ -146,7 +146,7 @@ Status ReorderOutputEdges(Graph* g, Node* n, int input_count,
     int dst_input = e->dst_input();
     g->RemoveEdge(e);
 
-    if (new_src_output < input_count - resource_input_count) {
+    if (new_src_output < static_cast<int64>(input_count - resource_input_count)) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/rearrange_function_argument.cc:149]: Better to extract RHS to a separate expression of proper type.

I am not sure there is an issue here though. All 3 variables have type `int`.

File: tensorflow/compiler/tf2xla/xla_compiler.cc [hunk 6]
```diff
@@ -1030,6 +1030,11 @@ Status XlaCompiler::BuildArguments(
       xla::XlaScopedShardingAssignment assign_sharding(
           builder, it == arg_shardings.end() ? absl::optional<xla::OpSharding>()
                                              : it->second);
+      auto& arg = args[input_to_args->at(i)];
+
+      xla::OpMetadata arg_metadata;
+      arg_metadata.set_op_name(arg.node_name);
+      builder->SetOneShotOpMetadata(arg_metadata);
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/xla_compiler.cc:1037]: Unrelated change

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Please remove unrelated changes

**Comment by tg-at-google** [comment]
updates applied.

File: tensorflow/compiler/tf2xla/functionalize_cond.cc (hunk 2 above)
Comment (mihaimaruseac) [reply tensorflow/compiler/tf2xla/functionalize_cond.cc:240]: Don't forget about these ones.

File: tensorflow/compiler/tf2xla/functionalize_cond.cc (hunk 3 above)
Comment (mihaimaruseac) [reply tensorflow/compiler/tf2xla/functionalize_cond.cc:248]: Don't forget about these ones.

File: tensorflow/compiler/tf2xla/functionalize_cond.cc (hunk 4 above)
Comment (mihaimaruseac) [reply tensorflow/compiler/tf2xla/functionalize_cond.cc:261]: Don't forget about these ones.

File: tensorflow/compiler/tf2xla/rearrange_function_argument.cc (hunk 5 above)
Comment (mihaimaruseac) [reply tensorflow/compiler/tf2xla/rearrange_function_argument.cc:149]: Don't forget about these ones.

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/tf2xla/rearrange_function_argument.cc (hunk 5 above)
Comment (tg-at-google) [reply tensorflow/compiler/tf2xla/rearrange_function_argument.cc:149]: updated.

**Comment by tg-at-google** [comment]
recommended updates applied.

File: tensorflow/compiler/tf2xla/functionalize_cond.cc [hunk 7]
```diff
@@ -235,14 +236,16 @@ StateMap::CondId StateMap::GetCondId(const StateMap::CondState& state) {
 }
 
 void StateMap::ResetCondId(const Node* node, StateMap::CondId id) {
-  if (node->id() < node_to_condid_map_.size())
+  const int64 node_to_map_size = node_to_condid_map_.size(); 
+  if (node->id() < node_to_map_size)
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/functionalize_cond.cc:240]:
```suggestion
  const int64 map_size = node_to_condid_map_.size(); 
  if (node->id() < map_size)
```

File: tensorflow/compiler/tf2xla/functionalize_cond.cc [hunk 8]
```diff
@@ -235,14 +236,16 @@ StateMap::CondId StateMap::GetCondId(const StateMap::CondState& state) {
...
     node_to_condid_map_[node->id()] = id;
   else
     added_node_condid_mapping_[node->id()] = id;
 }
 
 StateMap::AncestorId StateMap::LookupAncestorId(const Node* node) const {
-  if (node->id() < node_to_ancestorid_map_.size())
+  const int64 node_to_map_size = node_to_ancestorid_map_.size();
+  if (node->id() < node_to_map_size)
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/functionalize_cond.cc:248]:
```suggestion
  const int64 map_size = node_to_ancestorid_map_.size();
  if (node->id() < map_size)
```

File: tensorflow/compiler/tf2xla/functionalize_cond.cc [hunk 9]
```diff
@@ -254,7 +257,8 @@ StateMap::AncestorId StateMap::GetAncestorId(
 }
 
 void StateMap::ResetAncestorId(const Node* node, StateMap::AncestorId id) {
-  if (node->id() < node_to_ancestorid_map_.size())
+  const int64 node_to_map_size = node_to_ancestorid_map_.size();
+  if (node->id() < node_to_map_size)
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/functionalize_cond.cc:261]:
```suggestion
  const int64 map_size = node_to_ancestorid_map_.size();
  if (node->id() < map_size)
```

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Comment by tg-at-google** [comment]
updates applied, 3.

**Review by mihaimaruseac — APPROVED** [review]
